platforms/posix/src/px4_daemon/server.cpp

```diff
@@ -35,29 +35,27 @@
  *
  * @author Julian Oes <julian@oes.ch>
  * @author Beat Küng <beat-kueng@gmx.net>
+ * @author Mara Bos <m-ou.se@m-ou.se>
  */
 
+#include <errno.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <string.h>
 #include <string>
 #include <pthread.h>
 #include <poll.h>
 #include <sys/stat.h>
+#include <sys/socket.h>
 #include <sys/types.h>
+#include <sys/un.h>
+#include <vector>
 
 #include <px4_log.h>
 
 #include "pxh.h"
 #include "server.h"
 
-// In Cygwin opening and closing the same named pipe multiple times within one process doesn't work POSIX compliant.
```

Comment:
@m-ou-se **TOTALLY AWESOME!** 👍 👍  Thanks a lot for switching to sockets. I planned myself to attempt an implementation with this approach and did some research for that. Sockets have in my eyes only advantages for this use case, your commit message https://github.com/PX4/Firmware/commit/9594ebf72efdfff1a88aaf2206a90836ee9b9130 says it all. Also my main motivation would have been to get rid of this Cygwin hack (came in https://github.com/PX4/Firmware/pull/10371) to make it work under Windows. Thanks again, it works great on Windows now without any hack.

Reply:
Oh that's good news @MaEtUgR.